packages/macro/src/macroJs.ts

```diff
@@ -222,78 +207,80 @@ export default class MacroJs {
   processDescriptor = (descriptor_: Node) => {
     const descriptor = descriptor_ as ObjectExpression
 
-    this.types.addComment(descriptor, "leading", EXTRACT_MARK)
-    const messageIndex = descriptor.properties.findIndex(
-      (property) =>
-        isObjectProperty(property) && this.isIdentifier(property.key, MESSAGE)
-    )
-    if (messageIndex === -1) {
-      return descriptor
+    const messageProperty = this.getObjectPropertyByKey(descriptor, MESSAGE)
+    const idProperty = this.getObjectPropertyByKey(descriptor, ID)
+    const contextProperty = this.getObjectPropertyByKey(descriptor, CONTEXT)
+
+    const properties: ObjectProperty[] = [idProperty]
+
+    if (!this.stripNonEssentialProps) {
+      properties.push(contextProperty)
     }
 
     // if there's `message` property, replace macros with formatted message
-    const node = descriptor.properties[messageIndex] as ObjectProperty
+    if (messageProperty) {
+      // Inside message descriptor the `t` macro in `message` prop is optional.
+      // Template strings are always processed as if they were wrapped by `t`.
+      const tokens = this.types.isTemplateLiteral(messageProperty.value)
+        ? this.tokenizeTemplateLiteral(messageProperty.value)
+        : this.tokenizeNode(messageProperty.value, true)
+
+      let messageNode = messageProperty.value as StringLiteral
+
+      if (tokens) {
+        const messageFormat = new ICUMessageFormat()
+        const { message: messageRaw, values } = messageFormat.fromTokens(tokens)
+        const message = normalizeWhitespace(messageRaw)
+        messageNode = this.types.stringLiteral(message)
+
+        properties.push(this.createValuesProperty(values))
+      }
 
-    // Inside message descriptor the `t` macro in `message` prop is optional.
-    // Template strings are always processed as if they were wrapped by `t`.
-    const tokens = this.types.isTemplateLiteral(node.value)
-      ? this.tokenizeTemplateLiteral(node.value)
-      : this.tokenizeNode(node.value, true)
+      if (!this.stripNonEssentialProps) {
+        properties.push(
+          this.createObjectProperty(MESSAGE, messageNode as Expression)
+        )
+      }
 
-    let messageNode = node.value
-    if (tokens != null) {
-      const messageFormat = new ICUMessageFormat()
-      const { message: messageRaw, values } = messageFormat.fromTokens(tokens)
-      const message = normalizeWhitespace(messageRaw)
-      messageNode = this.types.stringLiteral(message)
+      if (!idProperty && this.types.isStringLiteral(messageNode)) {
+        // todo: context could be different types, get implementation from extractor
```

Comment:
Please act on this `todo`

Reply:
done, now it supports template literal as well.